cpp/tests/utilities/column_utilities.cu

```diff
@@ -308,9 +301,12 @@ struct column_comparator_impl<list_view, check_exact_equality> {
         return lhs_offsets[index] == rhs_offsets[index] ? false : true;
       });
 
-    // shrink back down
-    differences.resize(thrust::distance(differences.begin(), diff_iter));
-    print_differences(differences, lhs, rhs, print_all_differences, depth);
+    differences.resize(thrust::distance(differences.begin(), diff_iter));  // shrink back down
+
+    if (not differences.empty()) {
+      auto const msg = differences_message(differences, lhs, rhs, print_all_differences, depth);
```

Comment:
Why even create the variable `msg`? Could stream the output directly into FAIL...

Reply:
Good point :) Updated